### ISSUE TITLE
The "path" argument must be of type string. Received type undefined

### ISSUE THREAD

**Issue by moda20** [rkalis/truffle-plugin-verify#101]
i am having and issue when trying to verify that is the error : The "path" argument must be of type string. Received type undefined
" 
i don't know really what triggered it as the contract code is fairly standard, here is the debug version of the verification : 
```
Verifying contractX
Reading artifact file at /home/app/build/contracts/contractX.json
Retrieving constructor parameters from https://api-testnet.bscscan.com/api? .....
Constructor parameters retrieved: .......
Reading artifact file at /home/app/build/contracts/contractY.json
The "path" argument must be of type string. Received type undefined
Failed to verify 1 contract(s): contractX

```

the error is after reading the `contractY`. another thing is that this worked for me at first but when changing the environment to linux, this error appears. 

node : v10.19.0
Truffle : v5.4.3

**Comment by moda20** [comment]
IS this an issue with linked Libraries? because contracY is actually a Library that i link it to contractX when deploying;

**Comment by eugenekadish** [comment]
I am seeing the exact same issue with both Ropsten and test Binance Smart Chain. I am not using libraries.

Node: v16.6.0
Truffle: v5.4.3

**Comment by moda20** [comment]
update : i tried changing the version of this plugin to 5.1.3 but it had the same problem.

**Comment by BananaNeil** [comment]
I pushed a PR that solved this for me. Basically this plugin is not looking in your node_modules directory. The PR I pushed is pretty ugly, but it works. Hopefully someone who knows why it was written the way it was will have time to rethink how the node_modules directory should be scanned.

**Comment by rkalis** [comment]
Interesting, thanks for raising the PR @BananaNeil. Do you have a repo where you're experiencing this bug?, It doens't look like it applies to *all* projects with contracts from NPM.

**Comment by AlekseyMak** [comment]
Got the same error. 

My contract contains imports like these:


`import "../node_modules/@openzeppelin/contracts/access/AccessControlEnumerable.sol";
import "../node_modules/@openzeppelin/contracts/utils/Context.sol";
`

**Comment by rkalis** [comment]
Thanks @AlekseyMak! Do you (or any of the other people in this thread) have a repo that I can use to reproduce this bug?

**Comment by ctohster** [comment]
I'm having the same issue. Anyone fixed it?

**Comment by rkalis** [comment]
Thanks @dustinb for creating that repo, I'll try to look into it some time next week.

**Comment by rkalis** [comment]
Alright this should be fixed in v0.5.17. Please feel free to re-open if any issues persist. @BananaNeil thank you for the PR early on, but I ended up fixing it at the root of the issue.

This issue *only* happened when importing directly from `../node_modules`, for example:

```solidity
import "../node_modules/@openzeppelin/contracts/access/AccessControlEnumerable.sol";
```

The recommended way of importing npm packages is

```solidity
import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
```

Regardless, both methods should work now in v0.5.17!
